src/application/services/useNoteSettings.ts

```diff
@@ -65,13 +65,17 @@ export default function (): UseNoteSettingsComposableState {
   };
 
   /**
-   * Update note settings
+   * Update field isPublic in note settings
    *
    * @param id - Note id
-   * @param data - Note settings data with new values
+   * @param newIsPublicValue - new isPublic
    */
-  const update = async (id: NoteId, data: Partial<NoteSettings>): Promise<void> => {
-    noteSettings.value = await noteSettingsService.patchNoteSettingsByNoteId(id, data);
+  const updateIsPublic = async (id: NoteId, newIsPublicValue?: boolean): Promise<void> => {
+    if (noteSettings.value) {
```

Comment:
why do we need this statement?

Reply:
Because we can't use the construct like noteSettings.value?.isPublic = response.isPublic - expression may not be an optional property access

Reply:
your current code allows a case when you call `upadteIsPublic()` but nothing happened. It seems like a mistake, calling of `upadteIsPublic()` should always lead service call.